flare_portal/experiments/models/modules.py

```diff
@@ -0,0 +1,139 @@
+    @classmethod
+    def get_create_path_name(cls) -> str:
+        module_snake_case = cls.get_module_snake_case()
+        return f"{module_snake_case}_create"
+
+    @classmethod
+    def get_create_path(cls) -> str:
+        module_slug = cls.get_module_slug()
+        return (
+            "projects/<int:project_pk>/experiments/<int:experiment_pk>/modules/"
+            f"{module_slug}/add/"
+        )
+
+    @classmethod
+    def get_update_path_name(cls) -> str:
+        module_snake_case = cls.get_module_snake_case()
+        return f"{module_snake_case}_update"
+
+    @classmethod
+    def get_update_path(cls) -> str:
+        module_slug = cls.get_module_slug()
+        return (
+            "projects/<int:project_pk>/experiments/<int:experiment_pk>/modules/"
+            f"{module_slug}/<int:module_pk>/"
+        )
+
+    @classmethod
+    def get_delete_path_name(cls) -> str:
+        module_snake_case = cls.get_module_snake_case()
+        return f"{module_snake_case}_delete"
+
+    @classmethod
+    def get_delete_path(cls) -> str:
+        module_slug = cls.get_module_slug()
+        return (
+            "projects/<int:project_pk>/experiments/<int:experiment_pk>/modules/"
+            f"{module_slug}/<int:module_pk>/delete/"
+        )
+
+    def get_module_config(self) -> constants.ModuleConfigType:
+        raise NotImplementedError()
+
+    def get_module_description(self) -> str:
+        """Short description of module configuration"""
+        return ""
+
+    def __str__(self) -> str:
+        return f"PK: {self.pk} - Sort order: {self.sortorder}"
+
+
+class FearConditioningModule(BaseModule):
+    PHASES = Choices(
+        ("habituation", "Habituation"),
+        ("acquisition", "Acquisition"),
+        ("generalisation", "Generalisation"),
+        ("extinction", "Extinction"),
+        ("return_of_fear", "Return of fear"),
+    )
+    phase = models.CharField(max_length=24, choices=PHASES, default=PHASES.habituation)
+    trials_per_stimulus = models.PositiveIntegerField(default=0)
+    reinforcement_rate = models.PositiveIntegerField(default=0)
+    rating_delay = models.FloatField(default=1)
+    generalisation_stimuli_enabled = models.BooleanField(default=False)
+
+    def get_module_config(self) -> constants.ModuleConfigType:
+        return constants.ModuleConfigType(
+            id=self.pk,
+            type=self.get_module_tag(),
+            config={
+                "phase": self.phase,
+                "trials_per_stimulus": self.trials_per_stimulus,
+                "reinforcement_rate": self.reinforcement_rate,
+                "rating_delay": self.rating_delay,
+                "generalisation_stimuli_enabled": self.generalisation_stimuli_enabled,
+            },
+        )
+
+    def get_module_description(self) -> str:
+        details = [
+            self.get_phase_display(),
+            f"Trials per stimulus: {self.trials_per_stimulus}",
+            f"Reinforcement rate: {self.reinforcement_rate}",
```

Comment:
Could you point me towards the reinforcement rate again?

Reply:
Reinforcement rate is how many trials the US should play for the CS+. So if trials per stimulus is 12 and reinforcement rate is 12, then the US should play every time the CS+ is displayed. If reinforcement rate is 6, then the US should play 6 out of 12 times the CS+ is displayed.

Reply:
Thank you!